src/replication_manager.py

```diff
@@ -322,7 +340,7 @@ def get_ensured_sync_replica(self, replica_infos):
         return sync_quorum.get(helpers.get_oldest_replica(quorum_info))
 
 
-def _get_needed_replication_type(config, db, db_state, ha_replics):
+def _get_needed_replication_type_without_await_before_async(config, db, db_state, ha_replics):
```

Comment:
Ugh :(
But no better name comes to mind.
Maybe switch names with class method: leave here untouched and `_get_needed_replication_type_with_await_before_async` in mixin.

Reply:
Seems like pretty same solution

Reply:
> Seems like pretty same solution

Yep, not much better. But at least less negations.